docs/contribute_code_en.md

```diff
@@ -0,0 +1,126 @@
+# Contribute Code
+
+VisualDL is an open source project initiated by PaddlePaddle and ECharts.
+Our goal is to break through the deep learning black box and help the developers to see the training trend by providing visualization.
+We sincerely appreciate your contribution.  This document explains our workflow and work style.
+
+## Workflow
+
+VisualDL uses this [Git branching model](http://nvie.com/posts/a-successful-git-branching-model/).  The following steps guide usual contributions.
+
+1. Fork
+
+   To maintain the main repo and avoid dangling branches, please file Pull Requests from your fork. To make a fork, just head over to the GitHub page and click the ["Fork" button](https://help.github.com/articles/fork-a-repo/).
+
+1. Clone
+
+   To make a copy of your fork to your local computers, please run
+
+   ```bash
+   git clone https://github.com/your-github-account/VisualDL
+   cd VisualDL
+   ```
+
+1. Create the local feature branch
+
+   For daily works like adding a new feature or fixing a bug, please open your feature branch before coding:
+
+   ```bash
+   git checkout -b my-cool-stuff
+   ```
+
+1. Commit
+
+   Before issuing your first `git commit` command, please install [`pre-commit`](http://pre-commit.com/) by running the following commands:
+
+   ```bash
+   pip install pre-commit
+   pre-commit install
+   ```
+
+   Our pre-commit configuration requires [clang-format 3.8](http://releases.llvm.org/download.html) for auto-formating C/C++ code and yapf for Python.
```

Comment:
For some linux builds, they can also get it via "apt install clang-format-3.8" or "apt-get install clang-format-3.8"

Reply:
Also we use both yapf for auto format, and flake8 for python style checking (yapf does not do a good job of stripping unused imports and other PEP8 requirements).